buildkite/bazel-central-registry/bcr_postsubmit.py

```diff
@@ -23,24 +23,114 @@
   - Sync the bazel_registry.json and modules/ directory in the main branch of the BCR to https://bcr.bazel.build
 """
 
+import base64
+import hashlib
+import json
+import os
+import requests
 import subprocess
 import sys
 
 BCR_BUCKET = "gs://bcr.bazel.build/"
+ATTESTATION_METADATA_FILE = "attestations.json"
+FILES_WITH_ATTESTATIONS = ("source.json", "MODULE.bazel")
+
+
+class AttestationError(Exception):
+    """Raised when there is a problem wrt attestations."""
 
 def print_expanded_group(name):
     print("\n\n+++ {0}\n\n".format(name))
 
+def get_output(command):
+    return subprocess.run(
+          command,
+          encoding='utf-8',
+          stdout=subprocess.PIPE,
+      ).stdout
+
+def check_and_write_new_attestations():
+    print_expanded_group(":cop::copybara: Check & write attestations")
+    paths = get_new_attestations_json_paths()
+    if not paths:
+        # TODO: turn this into an error
+        print(f"No {ATTESTATION_METADATA_FILE} files were changed.")
+        return
+    
+    for p in paths:
+        check_and_write_module_attestations(p)
+
+def get_new_attestations_json_paths():
```

Comment:
If for some reason, a BCR commit didn't trigger the BCR postsubmit (e.g. failed webhook call), a follow up build won't pick up the missing commit, right? Should we store a last sync commit in the GCS bucket?

Reply:
Yeah, good point :( I'll come up with a fix.

Reply:
PTAL